### PR TITLE
PP-7255 Parity check task for refunds

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT YOU DID
- Currently we can parity check charges for a given ID range or parity check status. There is no way to parity check refund(s) using the task. For any fixes to refund parity checker we will have to wait until expunger picks the refund and parity checks the same.
Added functionality to parity check refunds by ID range and parity check status.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/tasks/ParityCheckWorker.java [hunk 1]
```diff
@@ -75,6 +84,27 @@ public void execute(Long startId, Optional<Long> maybeMaxId, boolean doNotReproc
...
+            } else {
+                maxId = ofNullable(maxId).orElseGet(refundDao::findMaxId);
+                processRefundsByIdRange(startId, maxId, doNotReprocessValidRecords);
+            }
+        } catch (Exception e) {
+            logger.error("Error parity checking refunds on job [start={}] [max={}] [error={}]",
+                    startId, maxId, e.getMessage(), e);
+        }
+        logger.info("Terminating");
+        MDC.remove(HEADER_REQUEST_ID);
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/tasks/ParityCheckWorker.java:105]: do we need to remove from the MDC for `execute` as well, if we're removing here?

File: src/main/java/uk/gov/pay/connector/tasks/ParityCheckWorker.java [hunk 2]
```diff
@@ -115,27 +145,81 @@ public void checkParityForIdRange(long startId, long maxId, boolean doNotReproce
     @Transactional
     public void checkParityFor(ChargeEntity charge, boolean doNotReprocessValidRecords) {
         try {
-            MDC.put("chargeId", charge.getExternalId());
+            MDC.put(PAYMENT_EXTERNAL_ID, charge.getExternalId());
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/tasks/ParityCheckWorker.java:148]: thanks for fixing that!

**Review by stephencdaly — APPROVED** [review]
Looks good. Question about MDC doesn't stop this PR being merged.

File: src/main/java/uk/gov/pay/connector/tasks/ParityCheckWorker.java (hunk 1 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/tasks/ParityCheckWorker.java:105]: yeah missed that. Will remove in the next PR

**Pull request merged**